src/revisions_parser.py

```diff
@@ -2,37 +2,120 @@
 from typing import Dict, List
 
 
-def parse(input: str) -> List[Dict[str, str]]:
-    # find start
-    iter = re.finditer(r"revision history|version control", input,
-                       re.IGNORECASE)
-    start = None
-    a = 0
-    for start in iter:
-        a += 1
-        # first mention probably in toc
-        if a == 2:
-            break
-    if not start:
-        return []
-    input = input[start.span(0)[1]:]
+def month_to_number(input: str) -> str:
+    values = {
+        "jan": "01",
+        "feb": "02",
+        "mar": "03",
+        "apr": "04",
+        "may": "05",
+        "jun": "06",
+        "jul": "07",
+        "aug": "08",
+        "sep": "09",
+        "oct": "10",
+        "nov": "11",
+        "dec": "12",
+    }
+    form = input[:3].lower()
+    if form in values:
+        return values[form]
+    return ""
+
+
+def clean_date(input: str) -> str:
+    if "-" in input:
+        splitted = input.split("-")
+        if len(splitted) != 3:
+            return input
+
+        if not splitted[1].isnumeric():
+            # swap year and day
+            tmp = splitted[0]
+            splitted[0] = splitted[2]
+            splitted[2] = tmp
 
-    # find end
+            # use number for months
+            splitted[1] = month_to_number(splitted[1])
+    elif "." in input:
+        splitted = input.split(".")
+        if len(splitted) != 3:
+            return input
+
+        # swap year and day
+        tmp = splitted[0]
+        splitted[0] = splitted[2]
+        splitted[2] = tmp
+    else:
+        return input
+
+    return "-".join(splitted)
+
+
+def parse_general(input: str, regex: str, ver_index: int, date_index: int) -> List[Dict[str, str]]:
     end = re.search(r"\n{4}", input, re.IGNORECASE)
     if end:
         input = input[:end.span(0)[1]]
     else:
         input = input[:5000]
-    
-    result = re.findall(r"\s+([0-9.]+)\s+([0-9-]+)?\s+(.*)", input)
-    
+
+    result = re.findall(regex, input)
+
+    final_result = []
     for i in range(len(result)):
```

Comment:
A more pythonic way of iteration is just `for res in result`. If naming the iterable in a way that its singular is identical causes problems (you would like `for result in ...`), consider adding an `s` or a name like `matches` for raw regex output, like I did. That way you also do not need to differentiate between `result` and `total_result`.